libcontainer/utils/utils_unix.go

```diff
@@ -364,3 +365,27 @@ func Openat(dir *os.File, path string, flags int, mode uint32) (*os.File, error)
 	}
 	return os.NewFile(uintptr(fd), dir.Name()+"/"+path), nil
 }
+
+// RetryOnEINTR takes a function that returns an error and calls it until it the error returned is
+// not EINTR.
+func RetryOnEINTR(fn func() error) error {
+	var err error
+	for {
+		err = fn()
+		if !errors.Is(err, unix.EINTR) {
+			break
+		}
+	}
+	return err
+}
+
+// RetryOnEINTR2 is like RetryOnEINTR, but returns 2 values: an int and an error.
+func RetryOnEINTR2(fn func() (int, error)) (fd int, err error) {
+	for {
+		fd, err = fn()
+		if !errors.Is(err, unix.EINTR) {
+			break
+		}
+	}
+	return fd, err
+}
```

Comment:
I think we can use generics for the second argument:

```go
 func RetryOnEINTR2[T any](fn func() (T, error)) (T, error) {
          for {
                  v, err := fn()
                  if !errors.Is(err, unix.EINTR) {
                          return v, err
                  }
          }
}
```

Reply:
Oh, right! Great idea, thanks!